src/NHibernate/Impl/AbstractSessionImpl.cs

```diff
@@ -31,26 +32,54 @@ public abstract partial class AbstractSessionImpl : ISessionImplementor
 
 		private bool closed;
 
+		/// <summary>Get the current NHibernate transaction.</summary>
+		public ITransaction Transaction => ConnectionManager.Transaction;
+
+		protected bool TransactionCoordinatorShared { get; }
```

Comment:
Yes, but no, and without any importance currently, that is why I have not really bothered.

For `SessionImpl`, serialization is explicitly done, causing fields from the base class to be serialized only if handled explicitly. It is not in `SessionImpl`. Anyway, if it was serialized, it would not change anything, because `GetObjectData` throws if it is not `false`, so a value other than the default one for its type would not be serialized.

For stateless, well, it could be now serialized, if the stateless was actually serializable, which is still not I believe. Stateless should implement serialization the same way than `SessionImpl`, meaning choosing itself what it serialize and if it can be serialized.